### ISSUE TITLE
Failure with dev testthat

### ISSUE THREAD

**Issue by hadley** [mlr-org/parallelMap#64]
I see:

*   checking tests ...
    ```
     ERROR
    Running the tests in ‘tests/test-all.R’ failed.
    Last 13 lines of output:
      3: suppressMessages({
             reg = BatchJobs::makeRegistry(id = basename(fd), file.dir = fd, work.dir = wd)
         }) at /Users/hadley/Documents/devtools/testthat/revdep/checks.noindex/parallelMap/new/parallelMap.Rcheck/00_pkg_src/parallelMap/R/parallelStart.R:173
      4: withCallingHandlers(expr, message = function(c) invokeRestart("muffleMessage"))
      5: BatchJobs::makeRegistry(id = basename(fd), file.dir = fd, work.dir = wd) at /Users/hadley/Documents/devtools/testthat/revdep/checks.noindex/parallelMap/new/parallelMap.Rcheck/00_pkg_src/parallelMap/R/parallelStart.R:174
      6: dbCreateJobStatusTable(reg)
      7: dbDoQuery(reg, query, flags = "rwc")
      8: stopf("Error in dbDoQuery. %s (%s)", res, query)
      
      ══ testthat results  ═══════════════════════════════════════════════════════════════════════════════
      OK: 62 SKIPPED: 0 WARNINGS: 288 FAILED: 1
      1. Error: warn on unregisterred level (@test_registerLevels.R#30) 
      
      Error: testthat unit tests failed
      Execution halted
    ```

There shouldn't be any breaking changes to testthat, so could you please confirm that this is a spurious failure? I'm aiming to submit to CRAN next week.

**Comment by jakob-r** [comment]
First I did not get any error, after installing r-lib/testthat I get the same error. Reverting back to cran:testthat and the error persists - so I am a bit confused. 

We are getting the warning `Don't need to call dbFetch() for statements, only for queries` from RSQLite a lot. This warning is converted to an error by calling `options(warn = 2L)` in the test that is now failing.

I cannot see how this should be introduced by testthat. It seems to me that we neglected this warning in the past. But this can be an issue to the devs of BatchJobs (where strangely this warning does not appear)
Pinging: @berndbischl @mllg
